docs/design/proposals/codegen-refactor/proposal-summary.md

```diff
@@ -0,0 +1,57 @@
+* `reconciliation`: logic involving access to or relations between `resource`, `shape`, `ackgenconfig`, and `aws-sdk`
+* `ackmodel`: the [code](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/model/model.go#L36) representation of ACK's view of the world; the source of truth for `aws-sdk`, `ackgenconfig`, and `reconciliation`
+
+
+## Problem
+*code-generator's* accelerated growth is resulting in tech debt that degrades the experience for platform maintainers and contributors. The codebase is becoming difficult to read and extend given the quick addition of features to onboard more services. This doc summarizes and introduces the proposals to eliminate significant tech debt so that development on ACK Platform is as pleasant and clear as it is open.
+
+
+## Solution
+
+### Overview
+To improve the implementation, I propose consolidating `ackconfig` access and any *ackconfig-aws-sdk* `reconciliation` into `ackmodel`. Taking this idea further, `ackmodel` should be the "source of truth"/required input for `api` and `controller` commands instead of *generator.yaml* and *aws-sdk*.
+
+
+### Requirements
+* Refactor does **not change** existing functionality
+* Code is easy to read, intuitive, and extendable
+* Code generation processes remain clear and transparent
+
+## Approach
+
+### Centralize config access and reconciliation
+By centralizing and consolidating config access and `reconciliation` logic, calls becomes significantly clearer. The diagrams compare the current and proposed function call paths:
+
+![current-config-access](./images/current_config_access.png)
+* `r` represents a `resource` or `crd`
+* `SetResource` calls `r.GetOutputShape(op)` to retrieve the shape for a given operation
+* Then, `r.GetOutputWrapperFieldPath(op)` is invoked; it's basically a wrapper for accessing `ackgenconfig`
+* Finally, `r.getWrapperOutputShape()` is called recursively and finally reconciles which shape to return
+* `r` is doing a lot of work and it isn't clear what is being resolved
+
+
+---
+
+![proposed-config-access](./images/proposed_config_access.png)
+* `m` represents `ackmodel`
+* `SetResource` calls `m.GetOutputShape(op)` now instead of `crd`
+* Under the hood, `ackmodel` will access its `ackgenconfig` to get config values
+* Then use helpers to reconcile between `sdk-shape` and `ackgenconfig`
+* With `m` being the source of truth and reconciliation logic, it is safe to assume the output shape you get back takes all configs into account
+* Also, `resource` is no longer bogged down with helpers unrelated to a `resource`
+
+
+### New command `./ack-generate model`
+With a new `model` command, the code generation can flow from a common data source which also creates opportunity for future commands:
```

Comment:
++  model command for dev workflow!

Reply:
Thanks for pointing this out, I'll make it clearer in next revision.

The output would be an [ackmodel](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/model/model.go#L36), serialized into JSON*, and stored in `~./cache/aws-controllers-k8s/ack-model.json`.

Something like:
```
./ack-generate model <params>
...
ACK model generated and stored successfully in ~./cache/aws-controllers-k8s/ack-model.json
```

Downstream commands will use default cache_dir to look for `ackmodel`. If none can be found, then error out requesting `ackmodel` to be built first.


* The current, proposed `ackmodel` as JSON would be huge! Experimenting locally and discussing offline with @RedbackThomson  to see how to break it up